packages/node_modules/samples/browser-multi-party-call/test/wdio/spec/normal-dialing.js

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable no-warning-comments */
+
+import '@ciscospark/plugin-phone';
+
+import CiscoSpark from '@ciscospark/spark-core';
+import testUsers from '@ciscospark/test-helper-test-users';
+import setValue from '../lib/set-value';
+
+describe(`sample-browser-multi-party-call`, () => {
+
+  describe(`normal dialing`, () => {
+    let mccoy, roomId, spock;
+
+    const browserSpock = browser.select(`browserSpock`);
+    const browserMccoy = browser.select(`browserMccoy`);
+
+    const users = {
+      mccoy: null,
+      spock: null
+    };
+
+    before(`create users and register`, () => testUsers.create({count: Object.keys(users).length})
+      .then((created) => Promise.all(Object.keys(users).map((name, index) => {
+        const user = users[name] = created[index];
+        user.spark = new CiscoSpark({
+          credentials: {
+            authorization: user.token
+          }
+        });
+
+        console.log(user.spark);
+
+        return user.spark.phone.register();
+      })))
+      .then(() => {
+        mccoy = users.mccoy;
+        spock = users.spock;
+      }));
+
+    beforeEach(`enable group calling`, () => {
+      spock.spark.config.phone.enableExperimentalGroupCallingSupport = true;
+      mccoy.spark.config.phone.enableExperimentalGroupCallingSupport = true;
+    });
+
+    afterEach(`disable group calling`, () => {
+      spock.spark.config.phone.enableExperimentalGroupCallingSupport = false;
+      mccoy.spark.config.phone.enableExperimentalGroupCallingSupport = false;
+    });
+
+    step(`create test room and add members to room`, () => {
+      spock.spark.request({
+        method: `POST`,
+        service: `hydra`,
+        resource: `rooms`,
+        body: {
+          title: `Call Test`
+        }
+      })
+        .then((res) => {
+          const room = res.body;
+
+          roomId = room.id;
+
+          return spock.spark.request({
+            method: `POST`,
+            service: `hydra`,
+            resource: `memberships`,
+            body: {
+              roomId: room.id,
+              personId: mccoy.id
+            }
+          });
+        });
+    });
+
+    step(`loads the app`, () => {
+      browser.reload();
+      browser.url(`/packages/node_modules/samples/browser-multi-party-call/`);
+    });
+
+    step(`connects mccoy's browser`, () => {
+      browserMccoy.waitUntil(() => browserMccoy.getTitle() === `Sample: Multi Party Calling`);
+      browserMccoy.execute(setValue, `[placeholder="Your access token"]`, mccoy.token.access_token);
+      // Debouncing before connecting ruduces flakiness
```

Comment:
What flakiness were you seeing?

Reply:
A common theme i have found with these tests is that the tests move a lot faster that the browser at times. In most cases if some pause isn't added (usually just 500ms does the trick), then the test fails with a lot of weird unexpected errors. The best, and most common example, is that I get error messages saying that the alert messages used to alert the browser of an incoming call weren't expect.

Reply:
Disregard the last comment, the flakiness of the test was due to the missing returns for the promise chains.